sample/redo/lib/main.dart

```diff
@@ -140,125 +142,122 @@ class _MyHomePageState extends State<MyHomePage> {
 
   Stream<Iterable<int>> _listIdStream() => _replicache.subscribe(
         (tx) async => (await tx.scan(prefix: '/list/')).map(
-          (item) => int.parse(item.key.substring('/list/'.length)),
+          (item) => item.value['id'],
         ),
       );
 
-  Stream<Iterable<Todo>> _todoStream() => _replicache.subscribe(
-        (tx) async => (await tx.scan(prefix: prefix)).map(
-          (item) => Todo.fromJson(item.value),
-        ),
-      );
+  Stream<Iterable<Todo>> _todoStream() => _replicache.subscribe(allTodosInTx);
 
   void _handleSync(bool syncing) {
     setState(() {
       _syncing = syncing;
     });
   }
 
-  Future<Todo> _read(ReadTransaction tx, int id) async {
+  static Future<Todo> _read(ReadTransaction tx, int id) async {
     final data = await tx.get(addPrefix(id));
     return data == null ? null : Todo.fromJson(data);
   }
 
-  Future<void> _write(dynamic tx, int id, Todo todo) {
-    return Future.error('Not implemented');
-    // return _replicache.put(addPrefix(key), todo.toJson());
+  static Future<void> _write(WriteTransaction tx, Todo todo) {
+    final key = addPrefix(todo.id);
+    return tx.put(key, todo);
+  }
+
+  static Future<bool> _del(WriteTransaction tx, int id) =>
+      tx.del(addPrefix(id));
```

Comment:
`'$prefix/$id'` is fewer chars than `addPrefix($id)`. Why have the helper ooc?

Reply:
Wondering if other people are going to want this, and if we should have an optional `prefix` field on `get` `put` `has`.

Reply:
I still think it is cleaner to have a function.

Reply:
OK in that case, let's pull it into the SDK? Put an optional `prefix` field on the relevant methods?